Redesign Licenses Screen

kevinrpb opened the pull request:
I thought the TextView looked a bit odd with the rest of the design. I'm starting to refactor it a bit. So far I've put the licenses into a JSON (this also allows to manage them dynamically if they change; or even fetch them if appropriate). Using this, the Licenses navigation gets you to a table view that will display a line for each product.

[Here's a preview of the new screen](https://pictshare.net/gvjkyl.mp4)

kevinrpb commented:
@rileytestut any chance for feedback here when you have some time? =)

dlevi309 commented:
@kevinrpb I just saw this preview and I actually really enjoyed it (not to sound surprised, all i mean by that is "how can you improve the licensing screen?" was my expectation) but I really like it, are you a part of the discord? if you'd like i could forward this over to him.

kevinrpb commented:
@dlevi309 Thank you =) I am not part of it. Is it public?

dlevi309 commented:
@kevinrpb yeah man :) and it is not public, you would have to join the 5$ a month pledge on his patreon, he's very responsive on it and would probably love to see this directly from you, but id be happy to put this on the channel for you 🙂

dlevi309 commented:
@kevinrpb also, are you accepting freelance work by any chance? id like to continue a conversation with you on that if you can! :P

ivar1803 commented on AltStore/Settings/LicenseDetailViewController.swift:
```diff
@@ -0,0 +1,34 @@
+        }
+
+        navigationItem.title = product
+        copyrightLabel.text = copyright != "" ? copyright : "(no copyright line)"
```
Can we use copyright.isEmpty?

ivar1803 commented on AltStore/Settings/LicensesViewController.swift:
```diff
@@ -8,42 +8,81 @@
+
+        let url = URL(fileURLWithPath: path)
+
+        guard let data = try? Data(contentsOf: url), let json = try? JSONSerialization.jsonObject(with: data) as? [[String: String]] else {
```
I suggest to go with Codable protocol instead of JSONSerialization

ivar1803 commented on AltStore/Settings/LicensesViewController.swift:
```diff
@@ -8,42 +8,81 @@
+            break
+        default:
+            cell.style = .middle
+        }
```
Move this switch case comparison to separate function. Like
cell.style = getStyleFor(license: [String: String])

OscarGorog commented:
Would be great to get this approved, looks great!

C0braD3v commented:
Looks amazing! +1

kevinrpb commented:
I'm closing this PR in favor of #149 since I have updated my fork to follow Git Flow and moved the changes to a new branch. I also took @ivar1803 suggestions and applied them.
